keymanager/src/crypto/kdf.rs

```diff
@@ -182,14 +140,13 @@ impl Inner {
         self.cache.clear();
     }
 
-    fn derive_key(&self, kdf_custom: &[u8], seed: &[u8], xof_custom: &[u8]) -> Result<KeyPair> {
+    // Derive ephemeral or long-term keys from the given secret.
+    fn derive_keys(&self, secret: Secret, xof_custom: &[u8]) -> Result<KeyPair> {
         let checksum = self.get_checksum()?;
-        let mut secret = self.derive_secret(kdf_custom, seed)?;
 
         // Note: The `name` parameter for cSHAKE is reserved for use by NIST.
         let mut xof = CShake::new_cshake256(&[], xof_custom);
-        xof.update(&secret);
-        secret.zeroize();
```

Comment:
No more explicit zeroize?

Reply:
No, as secret is zeroized when dropped. I should probably change that this function accepts `Secret` instead of `&[u8]` to be more obvious.

Reply:
Previous comment was meant for `derive_secret`. In `derive_keys` we don't need explicit zeroize as parameter `secret` is zeroized when dropped.